### PR TITLE
Simple forms – Add form pages 21-0972 alternate signer

### PR DESCRIPTION
**Pull request by gmrabian**
## Summary

- Add pages to formConfig
- Add helper functions to get dynamic page titles
- Add constants for checkbox/radio questions
- Add page uischema and schema configs
- Add test data for easy local testing, and future e2e tests

## Related issue(s)

- https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/300

## Testing done

- Tested locally at `/alternate-signer`

## What areas of the site does it impact?
21-0972 non prod

### CODE REVIEW

File: src/applications/simple-forms/21-0972/pages/preparerQualifications1.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      'ui:options': {
+        updateSchema: formData => {
+          // TODO: figure out why this isn't populating correctly
+          return {
+            title: preparerQualificationsQuestionTitle(formData),
+          };
+        },
+        forceDivWrapper: true,
+        // TODO: build out detailed info for each option
+        labels: Object.values(preparerQualificationsOptions),
```
Comment (gmrabian) [inline src/applications/simple-forms/21-0972/pages/preparerQualifications1.js:19]: the sketch has more info for each checkbox. Any suggestions?
Screenshot 2023-06-26 at 12 03 22 PM

File: src/applications/simple-forms/21-0972/pages/veteranIdentificationInformation1.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  uiSchema: {
+    // TODO: customize labels
+    veteranHasFiledClaim: {
+      ...yesNoUI('Has the Veteran ever filed a VA claim?'),
+      'ui:options': {
+        labels: {
+          Y: 'Yes, the Veteran has filed a VA claim before',
+          N: 'No, the Veteran has never filed a VA claim',
+        },
+      },
```
Comment (gmrabian) [inline src/applications/simple-forms/21-0972/pages/veteranIdentificationInformation1.js:17]: added a ticket to backlog – v3 pattern isn't accepting custom labels

File: src/applications/simple-forms/21-0972/pages/veteranPersonalInformation.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  titleUI,
+} from 'platform/forms-system/src/js/web-component-patterns';
+
+/** @type {PageSchema} */
+export default {
+  uiSchema: {
+    rjsf: titleUI(
+      '',
+      'Next we’ll ask you about the Veteran connected to the {claimant} you are certifying for.',
+    ),
```
[the remark this answers is not in the document]
Comment (rhasselle-oddball) [reply src/applications/simple-forms/21-0972/pages/veteranPersonalInformation.js:16]: I think we want a new variant of titleUI here. descriptionUI?  or... some other export we could retrieve from that file

File: src/applications/simple-forms/21-0972/pages/claimantAddress.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import {
+  addressSchema,
+  addressUI,
+} from 'platform/forms-system/src/js/web-component-patterns';
+
+/** @type {PageSchema} */
+export default {
+  uiSchema: {
+    claimantAddress: addressUI({ omit: ['isMilitary'] }),
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/21-0972/pages/claimantAddress.js:9]: thinking about this usage, vs a separate export specifically for this use case.

**Comment by rhasselle-oddball** [comment]
I don't see any blocking changes, as a WIP, so looks good to me. Appreciate all the inline comments.

File: src/applications/simple-forms/21-0972/config/form.js [hunk 5]
```diff
@@ -90,17 +119,163 @@ const formConfig = {
...
+    },
+    claimantPersonalInformationChapter: {
+      title: formData => claimantPersonalInformationTitle(formData),
+      pages: {
+        claimantPersonalInformation: {
+          path: 'claimant-personal-information',
+          title: formData => claimantPersonalInformationTitle(formData),
+          // skip if claimant is the veteran
+          depends: formData =>
+            formData?.claimantIdentification !== claimantIdentificationKeys[0],
```
[the remark this answers is not in the document]
Comment (ndsprinkle) [reply src/applications/simple-forms/21-0972/config/form.js:178]: I'm really not sure about this one. Assuming the function you wrote was solid, I would expect it to work no problem and function just like `title`. `claimantIndentificationKeys` and all the related variables were also brought over to that helpers file?

File: src/applications/simple-forms/21-0972/pages/preparerQualifications1.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  uiSchema: {
+    preparerQualifications: {
+      'ui:widget': GroupCheckboxWidget,
+      'ui:options': {
+        updateSchema: formData => {
+          // TODO: figure out why this isn't populating correctly
+          return {
+            title: preparerQualificationsQuestionTitle(formData),
+          };
+        },
```
[the remark this answers is not in the document]
Comment (ndsprinkle) [reply src/applications/simple-forms/21-0972/pages/preparerQualifications1.js:16]: For `updateSchema` not updating the title, maybe try `updateUiSchema`? That may produce different results, although I think `title` in the `schema` is the same as `ui:title` in the `uiSchema`. There's also `updateSchemaAndData` and a few other functions in here: `src/platform/forms-system/src/js/state/helpers.js`. But you may be completely right and it is related to group checkbox. I don't know exactly how that functions and there may be some issue with the scope of the update functions or the level it's at. I'll have to experiment with group checkbox more

[the remark this answers is not in the document]
Comment (ndsprinkle) [reply src/applications/simple-forms/21-0972/pages/preparerQualifications1.js:16]: https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/state/helpers.js#L376

File: src/applications/simple-forms/21-0972/pages/preparerQualifications1.js (hunk 1 above)
Comment (ndsprinkle) [reply src/applications/simple-forms/21-0972/pages/preparerQualifications1.js:19]: I don't even see this design supported here: https://design.va.gov/storybook/?path=%2Fdocs%2Fcomponents-va-checkbox-group--default. So I don't know where the designer pulled that from. Maybe they got confused with Radio? https://design.va.gov/storybook/?path=%2Fdocs%2Fcomponents-va-radio--default#tile

File: src/applications/simple-forms/21-0972/config/helpers.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+export const preparerSigningReasonQuestionTitle = ({ formData } = {}) => {
+  const claimantIdentificationText = getClaimantIdentificationText(formData);
+  return `Why does the ${claimantIdentificationText} need you to sign on their behalf? (Check all that apply)`;
+};
+
+export const claimantExpectedInformationDescription = ({ formData } = {}) => {
+  const claimantIdentificationText = getClaimantIdentificationText(formData);
+  return `Now, we’ll ask you about the ${claimantIdentificationText} you’ll be certifing to sign for`;
```
Comment (ndsprinkle) [inline src/applications/simple-forms/21-0972/config/helpers.js:42]: Certifying typo? :)

File: src/applications/simple-forms/21-0972/config/form.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (gmrabian) [reply src/applications/simple-forms/21-0972/config/form.js:178]: resolved by passing formData as `{formData}` in argument

File: src/applications/simple-forms/21-0972/pages/preparerQualifications2.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+export default {
+  uiSchema: {
+    preparerSigningReason: {
+      'ui:widget': GroupCheckboxWidget,
+      'ui:options': {
+        updateSchema: formData => {
+          return {
+            title: preparerSigningReasonQuestionTitle(formData),
+          };
+        },
```
[the remark this answers is not in the document]
Comment (gmrabian) [reply src/applications/simple-forms/21-0972/pages/preparerQualifications2.js:17]: resolved by passing formData as `{formData}` in argument

File: src/applications/simple-forms/21-0972/pages/veteranPersonalInformation.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (gmrabian) [reply src/applications/simple-forms/21-0972/pages/veteranPersonalInformation.js:16]: changed to `'ui:description': formData => veteranDescriptionText(formData),`

File: src/applications/simple-forms/21-0972/pages/preparerQualifications1.js (hunk 1 above)
Comment (gmrabian) [reply src/applications/simple-forms/21-0972/pages/preparerQualifications1.js:18]: Discussing options with design

**Review by rhasselle-oddball — APPROVED** [review]

**Pull request merged**